netebpfext/net_ebpf_ext_xdp.c

```diff
@@ -562,6 +563,7 @@ _net_ebpf_ext_handle_xdp_tx(
 
     if (status != STATUS_SUCCESS) {
         NET_EBPF_EXT_LOG_NTSTATUS_API_FAILURE(NET_EBPF_EXT_TRACELOG_KEYWORD_XDP, "FwpsInjectMacSendAsync", status);
+        _net_ebpf_ext_l2_inject_send_complete((void*)(uintptr_t)cloned_packet, nbl, FALSE);
```

Comment:
Shouldn't this pass through the dispatch flag?

Reply:
You're right.

Originally, I didn't spot a handy "is at dispatch level" flag here, and the meaning of a dispatch level flag in networking is conventionally "TRUE only if known to be at dispatch level" rather than "TRUE if and only if at dispatch level". However, WFP is sometimes a bit different (if the current IRQL is known, why not design the API to simply provide the current IRQL?) and I assumed the conventional rather than WFP meaning here, which was incorrect.

However, now that you've made me look more closely at this function, it appears the `FwpsFreeCloneNetBufferList` call above is passing an incorrect parameter: the second parameter is a reserved flags field, not a dispatch level boolean. Since the dispatch level flag is therefore redundant in practice, we can fix this by either adding an IRQL check (i.e., passing the correct value) or adding one layer of indirection to the cleanup routine. Since this is an edge case and we're not micro-optimizing performance in this driver, I'll go with querying the current IRQL.